Common/Securities/NullBuyingPowerModel.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IBuyingPowerModel"/> that uses an absurdly low margin
+    /// requirement to ensure all orders have sufficient margin provided the portfolio is not underwater.
+    /// </summary>
```

Comment:
Cool!
The only doubt with this guy is `SetHoldings` behavior because it will return a big number of holdings for say buying power of 1, since it uses the margin requirement and will still enforce margin requirements, it's just quantities are going to be **big**. See `PortfolioTarget.Percent()-> GetMaximumOrderQuantityForTargetBuyingPower`, maybe `GetMaximumOrderQuantityForTargetBuyingPower` should throw an invalid operation exception or reduce the quantity size returned somehow?

Reply:
Intersting. I did consider set holdings but appears I overlooked how it actually works. Maybe for the get max functions we do things more-or-less normally?